sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/autorestTest.java

```diff
@@ -0,0 +1,364 @@
+            System.out.print(e);
+        }
+
+    }
+
+
+    static AzureTableImpl auth() {
+        final String connectionString = System.getenv("azure_tables_connection_string");
+
+        StorageConnectionString storageConnectionString
+            = StorageConnectionString.create(connectionString, new ClientLogger("tables"));
+
+        StorageAuthenticationSettings authSettings = storageConnectionString.getStorageAuthSettings();
+        TablesSharedKeyCredential sharedKeyCredential = new TablesSharedKeyCredential(authSettings.getAccount().getName(),
+            authSettings.getAccount().getAccessKey());
+
+        final List<HttpPipelinePolicy> policies = Arrays.asList(
+            new AddDatePolicy(),
+            new AddHeadersPolicy(new HttpHeaders().put("Accept", OdataMetadataFormat.APPLICATION_JSON_ODATA_MINIMALMETADATA.toString())),
+            new TablesSharedKeyCredentialPolicy(sharedKeyCredential),
+            new HttpLoggingPolicy(new HttpLogOptions()
+                .setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+        );
+        final HttpPipeline pipeline = new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .build();
+        AzureTableImpl azureTable = new AzureTableImplBuilder()
+            .pipeline(pipeline)
+            .version("2019-02-02")
+            .url("https://telboytrial.table.core.windows.net")
+            .buildClient();
+        return azureTable;
+    }
+
+
+    @BeforeEach
+    void beforeTests() {
+        createTableHelper(tableA);
+        createTableHelper(tableB);
+
+        propertiesB.put("PartitionKey", "Store");
+        propertiesB.put("RowKey", "Boston");
+        propertiesB.put("Employees", "200");
+        insertEntityHelper(tableA, propertiesB);
+
+        Map<String, Object> propertiesA = new HashMap<>();
+        propertiesA.put("PartitionKey", "Store");
+        propertiesA.put("RowKey", "Atlanta");
+        propertiesA.put("Employees", "50");
+        insertEntityHelper(tableA, propertiesA);
+
+
+    }
+
+    @AfterEach
+    void afterTests() throws InterruptedException {
+        deleteTableHelper(tableA);
+        deleteTableHelper(tableB);
+
+    }
+
+    void createTableHelper(String tableName) {
+        AzureTableImpl azureTable = auth();
+
+        TableProperties tableProperties = new TableProperties().setTableName(tableName);
+        String requestId = UUID.randomUUID().toString();
+
+        azureTable.getTables().createWithResponseAsync(tableProperties, requestId,
+            ResponseFormat.RETURN_CONTENT, null, Context.NONE).block();
+
+    }
+
+    void deleteTableHelper(String tableName) throws InterruptedException {
+
+        AzureTableImpl azureTable = auth();
+        String requestId = UUID.randomUUID().toString();
+
+
+        azureTable.getTables().deleteWithResponseAsync(tableName, requestId,
+            Context.NONE).block();
+        TimeUnit.SECONDS.sleep(2);
```

Comment:
There's no need to sleep. The .block() call only moves to the next line when it is complete.